cider/exec/module/CiderRuntimeModule.cpp

```diff
@@ -179,40 +179,43 @@ CiderRuntimeModule::CiderRuntimeModule(
     , ciderCompilationOption_(ciderCompilationOption)
     , ciderExecutionOption_(ciderExecutionOption)
     , allocator_(allocator) {
-  hoist_literals_ = ciderCompilationOption.hoist_literals;
-  num_tables_ = 1;
-  error_code_ = 0;
-  literal_buff_ = ciderCompilationResult->getHoistLiteral();
-  build_table_ = std::move(ciderCompilationResult_->impl_->build_table_);
-  join_hash_tables_ =
-      ciderCompilationResult->impl_->compilation_result_.join_hash_table_ptrs;
-
-  auto& rel_alg_exe = ciderCompilationResult_->impl_->rel_alg_exe_unit_;
-
-  // pass concrete agg func to set different init_agg_vals
-  init_agg_vals_ = init_agg_val(rel_alg_exe->target_exprs,
-                                rel_alg_exe->simple_quals,
-                                *ciderCompilationResult_->impl_->query_mem_desc_,
-                                allocator_);
-
-  is_group_by_ = rel_alg_exe->groupby_exprs.size() && rel_alg_exe->groupby_exprs.front();
-  if (is_group_by_) {
-    initCiderAggGroupByHashTable();
-    initCiderAggTargetColExtractors();
-    group_by_agg_iterator_ = nullptr;
-  }
-
-  std::vector<TargetInfo> target_infos = target_exprs_to_infos(
-      rel_alg_exe->target_exprs, *ciderCompilationResult_->impl_->query_mem_desc_);
+  if (!ciderCompilationOption_.use_nextgen_compiler) {
```

Comment:
Hmm, for next gen, why not creating different instances separately? Like, we introduce a factory and instantiate impl. (either template-based or next-gen) via compilation option. It will help: (1) easily maintain two impl. moving forward; (2) easier to deprecate old template-based one, thoughts?